src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java

```diff
@@ -126,16 +128,29 @@ public String apply(String updatedDate, int currentIndex, Translation translatio
   SET_FIXED_LENGTH_DATA_ELEMENTS() {
     private transient DateTimeFormatter originCreatedDateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
     private transient DateTimeFormatter targetCreatedDateFormatter = DateTimeFormatter.ofPattern("yyMMdd");
-    private String fieldPattern = "%s|%s%s||||||||       |||||%s||";
+    private static final String DATES_OF_PUBLICATION = "datesOfPublication";
+    private static final String LANGUAGES = "languages";
+    private static final String FIELD_PATTERN = "%s|%s%s||||||||       |||||%s||";
 
     @Override
     public String apply(String originCreatedDate, int currentIndex, Translation translation, ReferenceData referenceData, Metadata metadata) {
-      String createdDateParam = targetCreatedDateFormatter.format(ZonedDateTime.parse(originCreatedDate, originCreatedDateFormatter));
+      String createdDateParam;
+      if (isNotEmpty(originCreatedDate)) {
+        try {
+          createdDateParam = targetCreatedDateFormatter.format(ZonedDateTime.parse(originCreatedDate, originCreatedDateFormatter));
+        } catch (DateTimeParseException e) {
+          LOGGER.error("Failed to parse createdDate field, the current time value will be used");
+          createdDateParam = targetCreatedDateFormatter.format(ZonedDateTime.now());
+        }
+      } else {
+        createdDateParam = targetCreatedDateFormatter.format(ZonedDateTime.now());
+      }
 
       String publicationDate0Param = "||||";
       String publicationDate1Param = "||||";
-      if (metadata != null && metadata.getData().containsKey("datesOfPublication")) {
-        List<String> publicationDates = (List<String>) metadata.getData().get("datesOfPublication").getData();
+      if (metadata != null && metadata.getData().containsKey(DATES_OF_PUBLICATION)
+        && metadata.getData().get(DATES_OF_PUBLICATION) != null) {
```

Comment:
I think something is incorrect here:
The publication field looks like this:
```
"publication" : [ {
      "publisher" : "American Bar Association",
      "place" : "Chicago, Ill.",
      "dateOfPublication" : null,
      "role" : null
    } ],
```

look at "dateOfPublication". There is no "date**s**OfPublication". 
Also, it returns an array, you should be checking if each entry is null or empty

Reply:
to be more clear, what if the metadata contains:   metadata.addData("datesOfPublication",  [null, 2016]);. Is this case handled?

Reply:
agree, not handled., will add more checks